modules/artifact/src/main/java/org/mule/module/artifact/classloader/ArtifactClassLoaderFilter.java

```diff
@@ -23,49 +25,72 @@
 public class ArtifactClassLoaderFilter implements ClassLoaderFilter
 {
 
-    private final ArtifactDescriptor descriptor;
+    public static final ArtifactClassLoaderFilter NULL_CLASSLOADER_FILTER = new ArtifactClassLoaderFilter(Collections.EMPTY_SET, Collections.EMPTY_SET);
 
-    public ArtifactClassLoaderFilter(ArtifactDescriptor descriptor)
+    public static final String EXPORTED_CLASS_PACKAGES_PROPERTY = "artifact.export.classPackages";
+    public static final String EXPORTED_RESOURCE_PACKAGES_PROPERTY = "artifact.export.resourcePackages";
+
+    private static final char PACKAGE_SEPARATOR = '.';
+    private static final String EMPTY_PACKAGE = "";
+    private static final char RESOURCE_SEPARATOR = '/';
+
+    private final Set<String> exportedClassPackages;
+    private final Set<String> exportedResourcePackages;
+
+    /**
+     * Creates a new classLoader filter
+     *
+     * @param exportedClassPackages class package names to export. Can be empty
+     * @param exportedResourcePackages resource package names to export. Can be empty
+     */
+    public ArtifactClassLoaderFilter(Set<String> exportedClassPackages, Set<String> exportedResourcePackages)
     {
-        this.descriptor = descriptor;
+        checkArgument(exportedClassPackages != null, "Exported class packages cannot be null");
+        checkArgument(exportedResourcePackages!= null, "Exported resource packages cannot be null");
+
+        this.exportedClassPackages = exportedClassPackages;
+        this.exportedResourcePackages = exportedResourcePackages;
     }
 
     @Override
-    public boolean accepts(String name)
+    public boolean exportsClass(String className)
     {
-        return !isBlockedClass(name) && isExportedClass(name) || !isBlockedPrefix(name) && isExportedPrefix(name);
-    }
+        final String packageName = getPackageName(className);
 
-    private boolean isBlockedPrefix(String name)
-    {
-        return hasListedPrefix(name, descriptor.getBlockedPrefixNames());
+        return exportedClassPackages.contains(packageName);
     }
 
-    private boolean isBlockedClass(String name)
+    @Override
+    public boolean exportsResource(String name)
     {
-        return descriptor.getBlockedPrefixNames().contains(name);
-    }
+        final String resourcePackage = getResourceFolder(name);
 
-    private boolean isExportedClass(String name)
-    {
-        return descriptor.getExportedPrefixNames().contains(name);
+        return exportedResourcePackages.contains(resourcePackage);
     }
 
-    private boolean isExportedPrefix(String name)
+    private String getResourceFolder(String resourceName)
     {
-        return hasListedPrefix(name, descriptor.getExportedPrefixNames());
+        if (resourceName == null)
+        {
+            return EMPTY_PACKAGE;
```

Comment:
Does this make sense, why would you get the resource folder for "null"?  Or is this simply preferred approach than failing with an exception?